File: isoft_android/linkknown/app/src/main/java/com/linkknown/ilearning/model/LoginUserResponse.java
package com.linkknown.ilearning.model;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper=true)
public class LoginUserResponse extends BaseResponse {

    private String domain;
    private long expireSecond;
    private String loginStatus;
    private String loginSuccess;
    private String nickName;
    private String redirectUrl;
    private String roleName;
    private String tokenString;
    private String userName;
    private String headerIcon;

    // 包含当前登录用户信息
    private UserDetailResponse userDetailResponse;
}

File: isoft_flutter/linkknown/lib/response/login_user_response.dart
import 'package:linkknown/response/base_response.dart';

class LoginUserResponse extends BaseResponse {
  String domain;
  String expireSecond;
  String headerIcon;
  String loginStatus;
  String loginSuccess;
  String nickName;
  String redirectUrl;
  String roleName;
  String tokenString;
  String userName;

  LoginUserResponse (
      {this.domain,
      this.expireSecond,
      this.headerIcon,
      this.loginStatus,
      this.loginSuccess,
      this.nickName,
      this.redirectUrl,
      this.roleName,
      this.tokenString,
      this.userName});

  LoginUserResponse.fromJson(Map<String, dynamic> json):
    domain = json['domain'],
    expireSecond = json['expireSecond'],
    headerIcon = json['headerIcon'],
    loginStatus = json['loginStatus'],
    loginSuccess = json['loginSuccess'],
    nickName = json['nickName'],
    redirectUrl = json['redirectUrl'],
    roleName = json['roleName'],
    tokenString = json['tokenString'],
    userName = json['userName'],
    super.fromJson(json);

}
